### ISSUE TITLE
docker volume rm does not delete to flocker volume in AWS

### ISSUE THREAD

**Issue by nderzhak** [ClusterHQ/flocker#2764]
In AWS VPC docker api call works fine with flocker volume drive:
```
docker volume create --driver=flocker --opt=size=${VOLUME_SIZE} --opt=profile=${VOLUME_PROFILE} --name=${VOLUME_NAME} 
```
And we can see created and attached volume in docker volume ls output.
```
flocker             pem-310-pre-prod-es-data-2
```
So we can see it in flockerctl list.
```
DATASET                                SIZE     METADATA                                       STATUS         SERVER                 
7d013471-e80c-4745-adc3-246b5a4d2a1d   10.00G   maximum_size=10737418240,name=pem-310-pre-     attached ✅   3bfaa4f5 (172.0.0.37)  
                                                prod-es-                                                                             
                                                data-1,clusterhq:flocker:profile=bronze                               
```
But docker volume rm ${VOLUME_NAME} simply does nothing.

So we have to use flockerctl destroy -d ${DATASET_ID} to accomplish deletion.

Is it something we can fix, ask to implement ?

**Comment by wallnerryan** [comment]
This is expected behavior right now actually. Before `docker volume ls` was moved to the responsibility of the drivers you would not see the volume, however now it is. I will speak to our team about changing this. thanks.

**Comment by jessesuen** [comment]
I also noticed this behavior where 'docker volume rm' does not work for flocker volumes. Unfortunately, it quite inconvenient for us to use flocker REST API to perform the deletion since it requires a completely separate authentication mechanism from the docker remote API and prevents us from abstracting the underlying docker storage driver.

Also, if deletion is not supported via 'docker volume rm', I expect the operation to raise some type of NotImplementedError rather than NOOP, since it gives a false sense of success.

**Comment by wallnerryan** [comment]
@jessesuen this was originally an intentional design. Since docker can receive many API requests and possibly across swarm nodes, we wanted to prevent accidental data loss by not "actually" deleting the volume but just the reference to the volume in docker. Since then, docker has moved the responsibility of LIST or `docker ls` to the drivers so you now see whether a volume which called `rm` is still there. We are discussing how to fix, and as your other issue suggested we welcome patches.

**Comment by jessesuen** [comment]
Based on my experiments in https://github.com/ClusterHQ/flocker/issues/2813, it appears docker-daemon maintains some type of in-memory state of volumes. Any out-of-band deletions of volumes confuses docker-daemon to the point where a volume name cannot be reused if the volume is deleted out-of-band. A docker-daemon restart will clear things up. Therefore, currently a user has no option but to perform **_both_** the 'docker volume rm <name>' immediately in conjunction with the flocker dataset deletion, in order to have a correct state. If the user need to performs both these steps anyways, it makes the most sense to have 'docker volume rm' behave according to spec.

**Comment by wallnerryan** [comment]
>  it makes the most sense to have 'docker volume rm' behave according to spec.

agreed, and we will see where this fits in with our roadmap, thanks for your input. cc @wallrj @myechuri

**Comment by wallnerryan** [comment]
https://clusterhq.atlassian.net/browse/FLOC-4377 should address this
